### ISSUE TITLE
OpenGD77 Support

### ISSUE THREAD

**Issue by consi** [sergev/dmrconfig#49]
Hey,

I'm using Baofeng DM-1801 with OpenGD77 (http://github.com/rogerclarkmelbourne/OpenGD77/) which is superb, but it's not being detected by dmrconfig (CPS utility for it looks similar on windows as original one.).

**Comment by n1zzo** [comment]
Hello,
I'm adding some details to help any developer who will tackle this issue.
The main difference between the stock firmware and OpenGD77 is the protocol used to transfer the CPS.
The stock fw uses probably a custom DFU variant, while OpenGD77 uses a text-based protocol
built on top of a USB virtual COM.
So the main issue would be to implement the virtual COM support, provided that no other radio
already has it. But on modern operating systems you directly get a serial tty, so it shoulndn't be hard.
The CPS layout should be quite identical, some bits which have undefined semantics in the
original fw have been assigned some behaviour in OpenGD77.

**Comment by rogerclarkmelbourne** [comment]
The OpenGD77 also supports 80 channel zones, but again, I don't think that its a big difference and would be easy to change in this code

**Comment by kzyapkov** [comment]
After some hesitation between improving the python script for OpenGD77 and this, I've started work on this. Looks like the serial protocol for OpenGD77 differs significantly from the one already implemented in `serial.c`, but I'll just export `dev_path` and implement it in `opengd77.c`.

@rogerclarkmelbourne can you please point me to some code or docs which explain the format of the binary blobs, and what goes where? Not particularly well versed in C# ...

**Comment by gwalborn** [comment]
Thanks. Looks promising!

Gary, WB8LEA

**Comment by kzyapkov** [comment]
Indeed, this was me on the forum and after some further consideration I've decided to go the python way. Speed and efficiency are not key here, a functioning CPS tool is. Also, this C code will need some rework in places to accommodate the new serial protocol.

**Comment by rogerclarkmelbourne** [comment]
OK

**Comment by naggie** [comment]
Thanks @kzyapkov ! Great to have that until support (hopefully) ends up in dmrconfig when someone has time.
